odl/set/sets.py

```diff
@@ -428,10 +428,10 @@ def __hash__(self):
 
     def element(self, inp=None):
         """Return a real number from ``inp`` or from scratch."""
-        if inp is not None:
-            return float(inp)
-        else:
+        if inp is None:
             return 0.0
+        else:
+            return float(getattr(inp, 'real', inp))
```

Comment:
Besides the implementation, I'm wondering if we really want this implicit conversion. I'd actually prefer to let this fail for objects that don't implement `__float__` but do implement `real`.
There's also a weird side effect for objects that implement both `__float__` and `real`, namely that `real` is preferred over `__float__`. That doesn't seem quite right to me.
Other opinions, @aringh, @adler-j?

Reply:
In this last case, should `ImagPart(RealNumbers().element(1))` give `0` or should it fail? The latter represents seeing `RealNumbers` and `ComplexNumbers` as two different entities, while the former somehow implicitly assumes that the real numbers are seen as embedded in the complex numbers, just with imaginary part 0. Which one is preferred?

Reply:
I'd take Python (and NumPy) as a guide here:
```python
>>> x = 1.0
>>> x.imag
0.0
>>> x = np.array(1.0)
>>> x.imag
array(0.)
>>> float(1 + 1j)
...
TypeError: can't convert complex to float
>>> np.array(1 + 1j, dtype=float)
...
TypeError: can't convert complex to float
```

Reply:
Some quick trials gives:

```
>>> (1).imag
0
```

Hence, `ImagPart(RealNumbers().element(1))` should return 0.